http3/body.go

```diff
@@ -1,84 +1,56 @@
 package http3
 
 import (
-	"fmt"
 	"io"
 
 	"github.com/lucas-clemente/quic-go"
+	"github.com/marten-seemann/qpack"
 )
 
+type trailerFunc func([]qpack.HeaderField, error)
+
 // The body of a http.Request or http.Response.
 type body struct {
-	str quic.Stream
+	str RequestStream
 
 	// only set for the http.Response
 	// The channel is closed when the user is done with this response:
 	// either when Read() errors, or when Close() is called.
 	reqDone       chan<- struct{}
 	reqDoneClosed bool
 
-	onFrameError func()
-
-	bytesRemainingInFrame uint64
+	onTrailers trailerFunc
```

Comment:
Is trailer parsing something we have to do in this PR? It would be really helpful to separate stuff like this into smaller, self-contained PRs.

Reply:
Trailers are not required for WebTransport, but were trivial to add using the new `http3.Conn` and `RequestStream` primitives. I can separate this out.